Feature and checkpoint review

kimani-ndegwa opened the pull request.

kimani-ndegwa committed:
(Chore) Moves to different branch for folder structure organisation

kimani-ndegwa committed:
(Chore) Achieves sound directory structure for project
(Chore) Creates User model and experiments with migrations

mrkimani-ndegwa committed:
(Chore) Reorganises file structure to suit project

mrkimani-ndegwa committed:
(Chore) Fixes bug with Api Blueprint in Flask

mrkimani-ndegwa committed:
(Chore) Migrations running well with User class only
(Feature) Adds context to shell migrations
(Chore) Updates git-ignore to prevent push of the databases as is.

mrkimani-ndegwa committed:
(Chore) Adds Swagger UI settings to config

kimani-ndegwa committed:
(Chore) Pulls from remote repo to get changes

kimani-ndegwa committed:
(Chore) Pulls from remote repo to get changes

kimani-ndegwa committed:
(Chore) Establishes model class User
(Chore) Makes tentatively final changes on folder structure

Merge branch 'Chore-folder-structure' of https://github.com/andela-kndegwa/CP2 into Chore-folder-structure

kimani-ndegwa committed:
(Chore) Adds tests andn functions

kimani-ndegwa committed:
(Chore) Adds Factory boy test

mrkimani-ndegwa committed:
(Chore) Adds all sqlite databases to the gitignore file to avoid push
(Chore) Icorporates Flask_Testing module
(Chore) Understands Namespaces in the flask_restplus domain

mrkimani-ndegwa committed:
(Chore) Updates requirements.txt file

kimani-ndegwa committed:
(Chore) Completes test_models module
(Chore) Code refactoring for purposes of more effective tests

kimani-ndegwa committed:
(Chore)Adds Namespaces to api config
(Feature) Can create bucket list and list all bucket lists

kimani-ndegwa committed:
(Chore) Confiigures restplus to work with manage.py

kimani-ndegwa committed:
(Chore) Leaves branch to migrate from Flask-RESTPlus to Flask- RESTFul

kimani-ndegwa committed:
(Chore) Succesfully moves from FLASK-RESTPlus to FLASK RESTFul
(Chore) Reorganises file structure to achieve the above move
(Feature) Register route works
(Feature) Login Route works
(Bug) Login is erratic

kimani-ndegwa committed:
(Chore) Fixes Bug with authentication to the API
(Chore) Refactors code-tests for the same

kimani-ndegwa committed:
(Feature) Authentication with secret key works properly

mrkimani-ndegwa committed:
(Chore)Ensures user can be added to bucketlist on creation and table.

kimani-ndegwa committed:
(Chore) CRUD Works well with Bucketlists

mrkimani-ndegwa committed:
(Chore) Standardizes REST Output of non-existent values to null
(Feature) Bucket list Item creation for API works.

kimani-ndegwa committed:
(Chore) Fixes bug with bucket list items

mrkimani-ndegwa committed:
(Chore) Adds code comments to the actions.py file
(Feature) Adds delete feature to the items resource

kimani-ndegwa committed:
(Chore) Refactors code to have all serializers on the the endpoint section
(Chore) BUCKET LIST-CRUD Functioning

kimani-ndegwa committed:
(Chore) refactors bucketlist item resource to reflect review changes

kimani-ndegwa committed:
(Chore) Updates requirements.txt file

kimani-ndegwa committed:
(Chore) Paginate feature implemenetation

mrkimani-ndegwa committed:
(Chore) Adds Pagination Feature to API and allows the API to filter out using the limit query string to url

mrkimani-ndegwa committed:
(Chore) Formats add_resource calls on the manage.py to adhere to PEP8
(Feature) Search bucket list feature implemented

kimani-ndegwa committed:
(Chore) Refactors tests and code to pass the tests
(BugFix) Eliminates creation of two clients for the testing classes.

mrkimani-ndegwa committed:
(Chore) Ensures tests are at a minimum 0f 73% coverage
(Chore) Adds .travis.yml file
(BugFix) Removes requried argument in the parser argument for bucketlists

mrkimani-ndegwa committed:
(Chore) Adds simplejson to requirements.txt file

mrkimani-ndegwa committed:
(Chore) Configures Tox wih project for py27
(Chore) Adds tox.ini file
(Chore) Adds travis badge to README.md

mrkimani-ndegwa committed:
(Chore) Adds python-coveralls to requirements.txt

mrkimani-ndegwa committed:
(Chore) Adds test coverage to 88%
(Chore) Configures tests to work with coveralls.io

mrkimani-ndegwa committed:
(Chore) Removes base_setup.py from fodler structure

mrkimani-ndegwa committed:
(Chore) Updates README.md file

mrkimani-ndegwa committed:
(Chore) Updates README.md file

mrkimani-ndegwa committed:
Merge branch 'feature-review' of https://github.com/andela-kndegwa/CP2 into feature-review
(Chore) Updates README.md file

mrkimani-ndegwa committed:
Merge branch 'feature-review' of https://github.com/andela-kndegwa/CP2 into feature-review

mrkimani-ndegwa committed:
Merge branch 'feature-review' of https://github.com/andela-kndegwa/CP2 into feature-review

mrkimani-ndegwa committed:
(Chore) Adds test coverage
(Chore) Updates README.md file
(Chore) Adds Circleci badge
(Chore) Adds screenshots to README.md file

mrkimani-ndegwa committed:
(Chore) Completes README.md file edit
(Fix) Ensures edit on bucketlists error handling.

kimani-ndegwa committed:
(Chore) Adds 404 Error handler.
(Chore) Removes unecessary import in __init__.py where app factory is

andela-kndegwa commented:
Adds comments to code on all the actions.py module.
Adds test coverage.
Uses pep8 style imports on the modules that import several functions from other modules.

Spaces out system imports with blank lines from local imports.

andela-kndegwa commented:
Configures the project to work with coveralls.io.

malikwahab commented on app/blister_api/actions.py:
```diff
@@ -0,0 +1,251 @@
+    if not data.get('title'):
+        abort(400, message='Please provide a title for your bucketlist')
+
+    title = data.get('title')
```
Get title before line 94, then you can use `title` for the `if` check to avoid calling the get statement twice.

malikwahab commented on app/blister_api/actions.py:
```diff
@@ -0,0 +1,251 @@
+        abort(400, message='Please add information for your bucketlist item.')
+    if not data.get('title'):
+        abort(400, message='Please provide a title for your bucketlist item')
+    title = data.get('title')
```
Do the same with title here as with line 94

kimani-ndegwa replied:
Cool.

kimani-ndegwa replied:
Am on it.

malikwahab reviewed (changes requested):
Looks Good to me.

kimani-ndegwa committed:
(Chore) Removes unnecessary files and folders from the .gitignore
(Chore) Makes changes as per peer review

kimani-ndegwa committed:
(Feature) Allows user to get a 200 Message even with no bucketlists

kimani-ndegwa committed:
(Feature) Allows user to get a 200 Message even with no bucketlists

andela-kndegwa commented:
@njirap Changes as recommended by @andela-aabdulwahab made and tests rerun.

kimani-ndegwa committed:
Merge branch 'feature-review' of https://github.com/andela-kndegwa/CP2 into feature-review

kimani-ndegwa committed:
(Chore) Changes lowercase base dir to upper case BASE_DIR as it is a global variable
(Chore) Ensures the imports have brackets around them if they are multiple
(Chore) allows shell context for bucketlist and bucketlist item in the manage.py

kimani-ndegwa committed:
(Chore) Disables error_404 suggestion when a 404 message is made.
(chore) Removes double call of the query item in the decorator and bucketlist pagination.

kimani-ndegwa committed:
(Chore) Removes double import in the items.py file
(Chore) Removes parenthesis on the User() model

andela-kndegwa commented:
Hi @njirap, Kindly see the latest commit with changes as suggested by Erika if you dont mind.

mrkimani-ndegwa committed:
(Chore) Adds codeclimate.yml file

mrkimani-ndegwa committed:
Merge branch 'feature-review' of https://github.com/andela-kndegwa/CP2 into feature-review
